src/lib/homestore_backend/hs_homeobject.cpp

```diff
@@ -198,29 +198,14 @@ void HSHomeObject::init_homestore() {
         svc_sb.create(sizeof(svc_info_superblk_t));
         svc_sb->svc_id_ = _our_id;
         svc_sb.write();
+        on_replica_restart();
```

Comment:
when we format homestore, it means this is the first-time start and no pg and shard should exist. so do we need this here?

Reply:
otherwise when we persist PG metadata we will get exception
```
[08/12/24 18:09:40-07:00] [C] [192263] [type=PGManager] not registered yet!
```

Reply:
so, can we only register pg and shard here., not do `read_sub_sb`?

Reply:
yes we can , but is that worth to saving 2 IOs?

Reply:
this is not nitpicking here and I do not want to save any io since no IO will happen(first-time start, metaservice will have not data for pg and shard).

 my point here is it seems a little weird to me that we have a `on_replica_restart` in the first-time start path, which should be in the recovery path.